ci-operator/step-registry/cucushift/ota/preupgrade/cucushift-ota-preupgrade-commands.sh

```diff
@@ -268,6 +268,106 @@ function pre-OCP-24358(){
+    token="$(oc -n openshift-monitoring create token prometheus-k8s)"
+    url="$(oc get route prometheus-k8s -n openshift-monitoring --no-headers|awk '{print $2}')"
+    
+    oc patch featuregate cluster --type json -p '[{"op": "add", "path": "/spec/featureSet", "value": "TechPreviewNoUpgrade"}]'
+    while (( retry > 0 )); do
+        reason="$(curl -s -k -H "Authorization: Bearer $token" "https://$url/api/v1/label/reason/values")"
+        if [[ "${reason}" == *"FeatureGates_RestrictedFeatureGates_TechPreviewNoUpgrade"* ]]; then
+            break
+        fi
+        sleep 1m
+        retry=$((retry-1))
+    done
+    if [[ "${reason}" != *"FeatureGates_RestrictedFeatureGates_TechPreviewNoUpgrade"* ]]; then
+        echo "Error: 32747 After waiting 10 minutes, FeatureGates_RestrictedFeatureGates_TechPreviewNoUpgrade still not appears"
+        return 1
+    fi
+
+    retry=5
+    while (( retry > 0 )); do
+        alert="$(curl -s -k -H "Authorization: Bearer $token" "https://${url}/api/v1/alerts" | jq  -r '.data.alerts[]| select(.labels.alertname == "ClusterNotUpgradeable")')"
+        namespace="$(echo "${alert}" | jq -r ".labels.namespace")"
+        severity="$(echo "${alert}" | jq -r ".labels.severity")"
+        description="$(echo "${alert}" | jq -r ".annotations.description")"
+        state="$(echo "${alert}" | jq -r ".state")"
+        if [[ "${namespace}" != "openshift-cluster-version" ]] \
+            || [[ "${severity}" != "info" ]] \
+            || [[ "${description}" != *"In most cases, you will still be able to apply patch releases."* ]] \
+            || [[ "${state}" != "pending" ]]; then
+            sleep 1m
+            retry=$((retry-1))
+            continue
+        fi
+        break
+    done
+    echo -e "Alert ClusterNotUpgradeable at beginning:\n${alert}"
+    if [[ "${namespace}" != "openshift-cluster-version" ]]; then
+        echo "Error: 32747 namespace is incorrect, expected is openshift-cluster-version, but observed is ${namespace}"
+        return 1
+    fi
+    if [[ "${severity}" != "info" ]]; then
+        echo "Error: 32747 severity is incorrect, expected is info, but observed is ${severity}"
+        return 1
+    fi
+    if [[ "${description}" != *"In most cases, you will still be able to apply patch releases."* ]]; then
+        echo "Error: 32747 description is incorrect, expected is 'In most cases, you will still be able to apply patch releases.', but observed is '${description}'"
+        return 1
+    fi
+    if [[ "${state}" != "pending" ]]; then
+        echo "Error: 32747 state is incorrect, expected is pending, but observed is ${state}"
+        return 1
+    fi
+    echo "Alert ClusterNotUpgradeable at beginning is correct"
+
+    sleep 50m   # the change only happened after 1 hour, so we can hardcode the sleep time here
+    echo "Sleep 50 minutes, then check the alert state"
+    retry=0
+    while (( retry < 40 ));do
```

Comment:
I don't heard of any changes or bugs on it, 90min seems not expected. Per [original checkpoint](https://gitlab.cee.redhat.com/aosqe/v4-installer-autotest/-/blob/master/cases/TestCases/OCP-32747/CVO_Alert_on_Upgradeable_False.py?ref_type=heads#L55-L67), the alert pending time should be 60min. But per your test result in https://github.com/openshift/release/pull/60108#issuecomment-2576494516, the time is 74m with a large deviation. Did you debug it locally or run the existed automation in v4-installer-autotest repo?

Reply:
I  followed the test case https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-32747, there isn't  much restriction on time duration. I think it's better to find out the feature which we followed to create the case.

@shellyyang1989 Could you let us know the background?

Reply:
> Wait for 1 hour, check the alert "ClusterNotUpgradeable"  again.

Hmm, there is waiting time described in last step. And from the Test Records, the case seems 100% passed from 4.15. So i guess 60mins in v4 automation repo should work well.

Reply:
OTA team is not so care about the waiting time, https://redhat-internal.slack.com/archives/CEGKQ43CP/p1737682127432489?thread_ts=1737632205.213959&cid=CEGKQ43CP, it's the responsibility of monitoring team. And there are many things can affect the wait time.

Reply:
I also asked the time in https://redhat-internal.slack.com/archives/C0VMT03S5/p1737682992614599, have not get a feedback.